File: app/src/main/java/com/codenicely/services/feelbazaar/vendorapp/splash/SplashActivity.java
package com.codenicely.services.feelbazaar.vendorapp.splash;

import android.content.Context;
import android.content.Intent;
import android.os.Handler;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import org.wikiedufoundation.wikiedudashboard.R;
import com.codenicely.services.feelbazaar.vendorapp.helper.SharedPrefs;
import com.codenicely.services.feelbazaar.vendorapp.home.HomeActivity;
import com.codenicely.services.feelbazaar.vendorapp.welcome.WelcomeActivity;

public class SplashActivity extends AppCompatActivity {

    private SharedPrefs sharedPrefs;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);
        final Context context= this;
        sharedPrefs = new SharedPrefs(context);
        Handler handler = new Handler();
        handler.postDelayed(new Runnable() {
            @Override
            public void run() {
            if (sharedPrefs.isLoggedIn()){
                startActivity(new Intent(context, HomeActivity.class));
                finish();
            }else {
                startActivity(new Intent(context,WelcomeActivity.class));
                finish();
            }
            }
        }, 1000);
    }
}

File: app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/splash/SplashActivity.kt
package org.wikiedufoundation.wikiedudashboard.ui.splash

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import androidx.appcompat.app.AppCompatActivity
import org.wikiedufoundation.wikiedudashboard.R
import org.wikiedufoundation.wikiedudashboard.data.preferences.SharedPrefs
import org.wikiedufoundation.wikiedudashboard.ui.home.HomeActivity
import org.wikiedufoundation.wikiedudashboard.ui.welcome.WelcomeActivity
import org.wikiedufoundation.wikiedudashboard.ui.welcome.onboarding.WelcomeHostActivity

/**
 * Splash screen activity
 * ***/
class SplashActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)

        val sharedPrefs = SharedPrefs(this)
        val handler = Handler()
        handler.postDelayed({
            if (sharedPrefs.isLoggedIn) {
                startActivity(Intent(this, HomeActivity::class.java))
                finish()
            } else {
                startActivity(Intent(this, WelcomeHostActivity::class.java))
                finish()
            }
        }, 1000)
    }
}
